src/com/google/ProductionInstrumentationReporter.java

```diff
@@ -0,0 +1,464 @@
+    while (line != null) {
+      sb.append(line).append("\n");
+      line = buf.readLine();
+    }
+    String fileAsString = sb.toString();
+    return fileAsString;
+  }
+
+  /**
+   * This function takes the instrumentationMapping and reports sent by the instrumented production
+   * code and creates a list of ProfilingResultsByFile with the aggregated information. It will
+   * average the ProfilingData provided by allInstrumentationReports and also calculate a percentage
+   * of how often a parameter is encounter.
+   *
+   * @param instrumentationMapping    The instrumentationMapping generated by the Production
+   *                                  Instrumentation pass
+   * @param allInstrumentationReports A list off all reports sent by the instrumented production
+   *                                  code
+   * @return A list of ProfilingResultsByFile where each element is the profiling result for a
+   * different source file.
+   */
+  private static List<ProfilingResultByFile> createProfilingResult(
+      InstrumentationMapping instrumentationMapping,
+      List<Map<String, ProfilingData>> allInstrumentationReports) {
+
+    List<ProfilingResultByFile> result = new ArrayList<>();
+
+    // Iterate over all fileNames since that is what we are grouping by.
+    for (String fileName : instrumentationMapping.fileNames) {
+
+      // Get all instrumentation parameters where the source file is fileName.
+      List<String> instrumentationPointsPerFile = instrumentationMapping.getAllMatchingValues((s) ->
+          fileName.equals(instrumentationMapping.getFileName(s))
+      );
+
+      Map<String, List<ProfilingResult>> profilingDataPerFunction = new HashMap<>();
+
+      for (String param : instrumentationPointsPerFile) {
+        String functionName = instrumentationMapping.getFunctionName(param);
+
+        float executionAverage = 0;
+        int averageFrequency = 0;
+        int runningAverageCounter = 0;
+
+        // For each param, iterate over allInstrumentationReports and check if param is present.
+        // If it is, we will average the data, otherwise we will add to the average as if it is 0.
+        for (Map<String, ProfilingData> instrumentationData : allInstrumentationReports) {
+          ProfilingData profilingData = instrumentationData.get(param);
+          runningAverageCounter++;
+          if (profilingData != null) {
+            // if executed, add 1 to executionAverage, and 0 otherwise.
+            averageFrequency += profilingData.frequency;
+            executionAverage++;
+          }
+        }
+
+        ProfilingResult profilingResult = new ProfilingResult(instrumentationMapping, param);
+        // Round the executionAverage to 2 decimal places for simplicity.
+        profilingResult.executed = roundFloat2Decimals(executionAverage / runningAverageCounter);
+        profilingResult.data = new ProfilingData(
+            Math.round((float) averageFrequency / runningAverageCounter));
```

Comment:
Why is it average? We don't have "total number of times this function was executed"?